### DISCUSSION TITLE
issue: isDirty is not properly updated when defaultValue is passed to useControlled

### DISCUSSION THREAD

**Discussion by csantos1113** [react-hook-form/react-hook-form#11494] [Q&A]
### Version Number

7.50.0

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/blazing-dew-smqxvw

### Steps to reproduce

1. Go to https://codesandbox.io/p/sandbox/blazing-dew-smqxvw
1. Notice "Cesar" is in:
    - the initial value of the input
    - the output of `watch()` at the bottom of the page
    - the submitted data if you click in "Submit"
1. Modify the input value. i.e. add character `'s'`
1. Check `isDirty` value:
    - `fieldState.isDirty` is true
    - `forState.isDirty` is true
1. Then, modify the input back to the original value: "Cesar". i.e. remove the character `'s'`)
1. 🤕 See error on `isDirty` values:
    - `fieldState.isDirty` is **still true**
    - `forState.isDirty` is **still true**

### Expected behaviour

I'd have expected the `isDirty` value to return back to false because the input value was returned to its original value

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Edge

### Additional information

it works as expected if I added `defaultValues` to `useForm` state, and remove it from `useControlled` (see [modified codesanbox](https://codesandbox.io/p/sandbox/fork-the-fork-86np7v)) and repeat all the steps

```diff
const formProps = useForm<FormValues>({
+  defaultValues: {
+    firstName: "Cesar",
+  },
});

const { field, fieldState } = useController<FormValues>({
-  defaultValue: "Cesar",
  name: "firstName",
});
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment]
The sandbox is not found.

**Comment by csantos1113** [comment]
> The sandbox is not found.

Thanks for the quick response. I didn't notice they were draft and private. I've updated the permissions and should work now!

**Comment by Moshyfawn** [comment]
I feel like it's the same story as #11402

https://github.com/react-hook-form/react-hook-form/issues/11402#issuecomment-1894433724

**Comment by csantos1113** [comment]
I guess the point that makes me believe they both should work the same is this section in the docs: https://react-hook-form.com/docs/usecontroller

<img width="1041" alt="image" src="https://github.com/react-hook-form/react-hook-form/assets/9648559/b2fdd0c0-110c-4c1e-996d-53d9960a6f14">
